src/Modules/OrchardCore.Commerce/Endpoints/Api/PaymentEndpoint.cs

```diff
@@ -23,10 +23,10 @@ public static IEndpointRouteBuilder AddFreeEndpoint(this IEndpointRouteBuilder b
 
     [Authorize(AuthenticationSchemes = "Api")]
     private static async Task<IResult> AddFreeAsync(
-         string? shoppingCartId,
-         IAuthorizationService authorizationService,
-         HttpContext httpContext,
-         IPaymentService paymentService
+         [FromRoute] string? shoppingCartId,
```

Comment:
Why not use them as properties decorated with `FromRoute`, `FromService` instead of injecting them within the constructor?

Reply:
This is a static class. How would that work?

Reply:
You can use it like this

```csharp
[HttpGet("api/products/{id}")]
public IActionResult GetProduct([FromRoute] int id)
{
    // id will be populated with the value from the route
}
```

Reply:
I remember I used [FromRoute] in the RazorPage model property

```csharp
[FromRoute("id")]
public string Id { get; set; }
```

Reply:
Could you tell me why this is a better approach? What's the benefit? @hishamco

Reply:
Much cleaner and no need to use constructor injection

Reply:
I think this could only be done by turning this from a minimal API endpoint class into an API controller? But as I recall OC is moving the other direction (away from API controllers) for performance reasons.